src/cas/anndata_to_cas.py

```
@@ -28,80 +32,105 @@ def anndata2cas(
 
     Returns:
         None
     """
-    if include_hierarchy:
-        for key, value in parent_cell_look_up.items():
-            for inner_key, inner_value in parent_cell_look_up.items():
-                if value != inner_value and value.get("cell_ids").issubset(
-                    inner_value.get("cell_ids")
-                ):
-                    value.update(
-                        {
-                            "parent": inner_key,
-                            "p_accession": inner_value.get("accession"),
-                        }
-                    )
-
-        annotation_list = cas.get("annotations")
-        for annotation in annotation_list:
-            annotation.update(
-                {
-                    "parent_cell_set_name": parent_cell_look_up.get(
-                        annotation.get("cell_label")
-                    ).get("parent")
-                }
-            )
-            annotation.update(
-                {
-                    "parent_cell_set_accession": parent_cell_look_up.get(
-                        annotation.get("cell_label")
-                    ).get("p_accession")
-                }
-            )
+    for (key, value), (inner_key, inner_value) in itertools.combinations(
+        parent_cell_look_up.items(), 2
+    ):
+        if value.get("parent") is not None and value.get(
+            "parent_rank", 0
+        ) <= inner_value.get("rank"):
+            continue
+
+        if value["cell_ids"] != inner_value["cell_ids"] and value["cell_ids"].issubset(
+            inner_value["cell_ids"]
+        ):
+            update_parent_info(value, inner_key, inner_value)
+        elif value["cell_ids"] == inner_value["cell_ids"]:
+            if int(inner_value["rank"]) < int(value["rank"]):
+                update_parent_info(value, key, value)
+            elif int(value["rank"]) < int(inner_value["rank"]):
+                update_parent_info(value, inner_key, inner_value)
+
```

Let's throw an exception ("cell_ids can't be identical at the same rank.") in `else`. This will significantly help us on identifying data related problems.